src/libraries/Common/src/System/Net/ArrayBuffer.cs

```diff
@@ -185,7 +185,7 @@ public void TryEnsureAvailableSpaceUpToLimit(int byteCount, int limit)
                 ArrayPool<byte>.Shared.Return(oldBytes);
             }
 
-            Debug.Assert(byteCount <= AvailableLength);
+            Debug.Assert(byteCount <= AvailableLength || ActiveLength + AvailableLength == limit);
```

Comment:
I believe that ArrayPool can actually give you a larger buffer than you requested (right?).

In that case this assert would not be correct. I'd suggest "desiredSize == limit" instead.

Reply:
> right?

Yup

> I'd suggest "desiredSize == limit" instead.

Ok.  I was planning on just replacing this with what you have in your PR, anyway.